### ISSUE TITLE
Error message not helpful when attempting to JSON parse the access token

### ISSUE THREAD

**Issue by timharsch** [smart-on-fhir/client-js#101]
I'm writing some new code, which potentially has errors and I'm trying to determine what's wrong with my code.  Whatever the issue it leads down the code path in fhirclient of attempting to parse the access token and fails with

> Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.

As you can see from my debugger:
![image](https://user-images.githubusercontent.com/61298249/95495171-df67f600-0953-11eb-939f-14f224736cea.png)

It's making the assumption that the access code contains a period.  It does not, so atob fails because it is passed undefined as a parameter.   I'm working with Epic Sandbox.  It would be helpful if that error message stated something more like "Access token invalid" or if, an even better error message that might me clue me in as to what I'm doing wrong.  Why is my token in invalid, if it is?  I'm not sure how I'm in the state where Epic has given me an invalid token.  Is there something that could be done leading up to this error even farther up the call stack that could help?

**Comment by vlad-ignatov** [comment]
Right! Thinking that the access token is always a JWT (which can be split on ".") was an incorrect assumption. I am working on a fix. Can you just verify that Epic is giving you the `expires_in` property as part of the token response?

**Comment by timharsch** [comment]
Obfuscated to satisfy my general paranoia, though being a sandbox I'm interacting with I'm sure it's overkill.   No periods were harmed in the making of this image...  :-)

![image](https://user-images.githubusercontent.com/61298249/95515336-4c8a8400-0972-11eb-9b3c-639de6f9a716.png)

**Comment by vlad-ignatov** [comment]
The fix is ready to be released. Before I do that, if you are using this as a library (via script tag), can you switch to the latest build at https://combinatronics.com/smart-on-fhir/client-js/master/dist/build/fhir-client.js to verify that it works properly for you?

**Comment by timharsch** [comment]
I'm in a React project and I can't test in browser without rewiring things a bit, but I'll see if I can find a way to get yarn to include a github module from commit, or I will pull the repo and install locally.  Got my morning meetings coming up so might be a couple of hours yet.

**Comment by timharsch** [comment]
Doing the yarn method didn't seem to work. `yarn add ssh://git@github.com:smart-on-fhir/client-js.git#e1d61bcf21a31ac28642b016326003e9f6e8e819`   It imports with the add command just fine, and then yarn install works, but when I launch with yarn start I get `Module not found: Can't resolve 'fhirclient'`

**Comment by timharsch** [comment]
Sorry, guess I can't find a way to test.  I tried using yarn link this time.  My workflow to try that basically looks like this

```
# git clone ssh://git@github.com:smart-on-fhir/client-js.git

# mv client-js fhirclient; cd fhirclient

# git checkout e1d61bc

# npm install

# npm run build

# yarn link
yarn link v1.22.4
success Registered "fhirclient".
info You can now run `yarn link "fhirclient"` in the projects where you want to use this package and it will be used instead.
✨  Done in 0.05s.

# cd ../myproject

# yarn link fhirclient
yarn link v1.22.4
success Using linked package for "fhirclient".
✨  Done in 0.05s.

<change my package.json to include fhirclient:2.3.9>

# yarn install
Couldn't find any versions for "fhirclient" that matches "2.3.9"
? Please choose a version of "fhirclient" from this list: (Use arrow keys)
```

Do you have any other ideas?  If not, perhaps you can just release your code and I'll test via the normal path, I noticed that your commit includes some good tests so maybe that's sufficient

**Comment by vlad-ignatov** [comment]
OK. Thank you very much for trying. I pushed new dev release to npm. You can install it with whatever is the yarn equivalent of `npm i fhirclient@dev` or `npm i fhirclient@2.3.9`.

**Comment by timharsch** [comment]
Your welcome!  Thank you for the quick response and addressing the issue.

I can confirm that the code no longer throws the confusing atob error but better yet that I can authenticate and download records from Epic sandbox now! 💯  I also tested against Cerner sandbox and can confirm it is still fetching records successfully as well.

I did encounter an unrelated issue though because I attempted to refresh my window where I downloaded from Cerner, that was hours old, and it ran into the missing CORS Access-Control-Allow-Origin header.  Looks like it is only for refresh token.  Don't know if I can repro that again..  I know it was discussed in issue #97.  I don't have much to go on but I'll file an issue if I see the problem come up again.

**Comment by vlad-ignatov** [comment]
Thanks @timharsch
The CORS issue is very concerning. Cerner should be fixing that on their end and the client should have already been fixed to work regardless of the missing header. The only way to reproduce that would be to let a page stay idle for more than 10 min (or whatever the access token lifetime is) and then refresh. However, I really hope that it was some kind of browser cache issue, although not very likely.

**Comment by vlad-ignatov** [comment]
OK, this is now included in the latest official release - `2.3.10`. If you encounter that CORS problem again, please open separate issue. Thank you!
